src/components/calculators/show.js

```diff
@@ -48,20 +50,40 @@ export class CalculatorShow extends Component {
 
   onChange({id, guesstimate}, input) {
     const guesstimateType = Guesstimator.parse({...guesstimate, input})[1].samplerType().referenceName
+    const shouldRunSims = !_.isEmpty(input) && this.allInputsHaveContent([id])
+    const shouldRunAllUnsimulated = !this.state.hasSimulated
 
     this.props.changeGuesstimate(
       id,
       {...guesstimate, ...{data: null, input, guesstimateType}},
-      true, // runSims
-      false // saveOnServer
+      shouldRunSims, // runSims
+      false, // saveOnServer
+      shouldRunAllUnsimulated // runAllUnsimulated
     )
+    if (shouldRunSims) {this.setState({hasSimulated: true})}
   }
 
-  allInputsHaveContent() {
-    const inputComponents = _.map(this.props.inputs, metric => this.refs[`input-${metric.id}`])
+  onEnter() {
+    if (!this.state.showResult && this.readyToCalculate()) {
+      this.setState({showResult: true})
+    }
+    return true
```

Comment:
I feel like this shouldn't have to know about the 'return true' statement.  That one element should be reserved to the input.